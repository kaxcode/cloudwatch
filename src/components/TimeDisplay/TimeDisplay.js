import { millisecondsToHuman } from '../../utils/humanizeTimer';
import { number, oneOfType, string } from 'prop-types';
import React from 'react';

const TimeDisplay = ({time}) => (
  <h3 className="Stopwatch__Count">
    {millisecondsToHuman(time)}
  </h3>
);

TimeDisplay.propTypes = {
  time: oneOfType([number, string])
};

export default TimeDisplay;
