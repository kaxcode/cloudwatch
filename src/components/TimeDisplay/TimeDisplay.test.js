import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-localstorage-mock';
import TimeDisplay from './TimeDisplay';

Enzyme.configure({ adapter: new Adapter() });

describe('TimeDisplay', () => {
  it('renders correctly', () => {
    expect(toJson(shallow(<TimeDisplay time={'20000'} />))).toMatchSnapshot();
  });
});
