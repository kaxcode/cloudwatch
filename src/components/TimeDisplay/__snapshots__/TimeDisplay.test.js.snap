// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h3
  className="Stopwatch__Count"
>
  00:00:20:000
</h3>
`;
